### PR TITLE
Error message for wrong file type refactored.

### PR DESCRIPTION
**Pull request by OgunyemiO**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EXUI-1038

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -283,7 +283,16 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
     }
-    return error.error;
+
+    let errorMsg = null;
+    if (error && error.error) {
+      const fullError = error.error;
+      const start = fullError.indexOf('{');
+      const json = fullError.substring(start, fullError.length - 1).split('<EOL>').join('');
+      const obj = JSON.parse(json);
+      errorMsg = obj.error;
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:293]: if (obj && obj?.error) {
  errorMsg = obj.error
}

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -283,7 +283,16 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
     if (0 === error.status || 502 === error.status) {
       return WriteDocumentFieldComponent.UPLOAD_ERROR_NOT_AVAILABLE;
     }
-    return error.error;
+
+    let errorMsg = null;
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:287]: let errorMsg = 'Error uploading file';

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 3]
```diff
@@ -283,7 +283,16 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+    if (error && error.error) {
+      const fullError = error.error;
+      const start = fullError.indexOf('{');
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:290]: Need to check that the error message actually contains some json:
if (start >= 0) {

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Need to improve the error handling

**Comment by OgunyemiO** [comment]
> Need to improve the error handling

@andywilkinshmcts Done!

**Review by olusegz07 — DISMISSED** [review]

**Review by andywilkinshmcts — DISMISSED** [review]

**Pull request merged**